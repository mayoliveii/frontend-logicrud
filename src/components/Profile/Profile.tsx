import { Badge, Flex, Image, Text, useColorModeValue } from "@chakra-ui/react";

interface ProfileProps {
  backgroundImage: string;
  profileImage: string;
  name: string;
  role: string;
  description: string;
  tags: string[];
}

export function Profile({
  backgroundImage,
  profileImage,
  name,
  role,
  description,
  tags
}: ProfileProps) {
  const boxBg = useColorModeValue("white !important", "#111c44 !important");
  const mainText = useColorModeValue("gray.800", "white");
  const secondaryText = useColorModeValue("gray.400", "gray.400");

  return (
    <Flex
      borderRadius='20px'
      bg={boxBg}
      p='20px'
      h='400px'
      w={{ base: "315px", md: "345px" }}
      alignItems='center'
      boxShadow='dark-lg'
      direction='column'
      justifyContent='space-between'
    >
      <Flex flexDirection='column'>
        <Image
          src={backgroundImage}
          w='300px'
          h='110px'
          borderRadius='20px'
        />
        <Flex flexDirection='column' mb='10px'>
          <Image
            src={profileImage}
            border='5px solid red'
            mx='auto'
            borderColor={boxBg}
            width='68px'
            height='68px'
            mt='-38px'
            borderRadius='50%'
          />
          <Text
            fontWeight='600'
            color={mainText}
            textAlign='center'
            fontSize='xl'>
            {name}
          </Text>
          <Text
            color={secondaryText}
            textAlign='center'
            fontSize='sm'
            fontWeight='500'>
            {role}
          </Text>
        </Flex>
        <Text
          color={mainText}
          fontSize='md'
          textAlign='center'
          mb='10px'>
          {description}
        </Text>
      </Flex>
      <Flex wrap="wrap" justifyContent="center">
        {tags.map((tag) => (
          <Badge colorScheme="gray" fontSize="sm" m="1">{tag}</Badge>
        ))}
      </Flex>
    </Flex>
  );
}