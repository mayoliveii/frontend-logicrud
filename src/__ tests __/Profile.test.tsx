import { render } from '@testing-library/react';
import { ChakraProvider } from '@chakra-ui/react';
import { Profile } from '../components/Profile/Profile';
import '@testing-library/jest-dom';

const profileProps = {
  backgroundImage: 'testBackground.jpg',
  profileImage: 'testProfile.jpg',
  name: 'John Doe',
  role: 'Engineer',
  description: 'Engineer with 5 years of experience',
  tags: ['tag1', 'tag2', 'tag3']
};

describe('Profile', () => {
  it('renders Profile component and matches snapshot', () => {
    const { asFragment } = render(
      <ChakraProvider>
        <Profile {...profileProps} />
      </ChakraProvider>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders Profile component with given props', () => {
    const { getByText } = render(
      <ChakraProvider>
        <Profile {...profileProps} />
      </ChakraProvider>
    );

    expect(getByText(profileProps.name)).toBeInTheDocument();
    expect(getByText(profileProps.role)).toBeInTheDocument();
    expect(getByText(profileProps.description)).toBeInTheDocument();
    profileProps.tags.forEach(tag => {
      expect(getByText(tag)).toBeInTheDocument();
    });
  });
});