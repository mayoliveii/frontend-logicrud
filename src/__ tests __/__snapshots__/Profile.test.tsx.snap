// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile renders Profile component and matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="css-1mhvcxt"
  >
    <div
      class="css-j7qwjs"
    >
      <img
        class="chakra-image css-g1rp98"
        src="testBackground.jpg"
      />
      <div
        class="css-9jmnaf"
      >
        <img
          class="chakra-image css-r3tnns"
          src="testProfile.jpg"
        />
        <p
          class="chakra-text css-13ufrgq"
        >
          John Doe
        </p>
        <p
          class="chakra-text css-3jgsvj"
        >
          Engineer
        </p>
      </div>
      <p
        class="chakra-text css-mufhc5"
      >
        Engineer with 5 years of experience
      </p>
    </div>
    <div
      class="css-7m4n47"
    >
      <span
        class="chakra-badge css-aene3b"
      >
        tag1
      </span>
      <span
        class="chakra-badge css-aene3b"
      >
        tag2
      </span>
      <span
        class="chakra-badge css-aene3b"
      >
        tag3
      </span>
    </div>
  </div>
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;
